types/types.go

```diff
@@ -181,10 +181,9 @@ type NodeConfig struct {
 	Memory string  `json:"memory,omitempty"`
 
 	// Extra node parameters
-	Extras  *Extras    `json:"extras,omitempty"`
-	WaitFor []string   `json:"wait-for,omitempty"`
-	DNS     *DNSConfig `json:"dns,omitempty"`
-
+	Extras  *Extras                     `json:"extras,omitempty"`
+	WaitFor map[WaitForPhase][]*WaitFor `json:"wait-for,omitempty"`
```

Comment:
and similar to https://github.com/srl-labs/containerlab/pull/1759/files#r1464920202 I think this should rather be `Stages`, as WaitFor is now part of the stage, and not directly a property of a Node